src/modules/roc_fec/idecoder.h

```diff
@@ -23,14 +23,24 @@ class IDecoder {
 public:
     virtual ~IDecoder();
 
+    //! Perform an initial setup for current block:
+    //!  * Set number of source packets and repair packets.
+    //!
+    //! @note
+    //!  Should be called as a first decoder operation for current block.
+    virtual bool begin(size_t sblen, size_t rblen) = 0;
+
     //! Store source or repair packet buffer for current block.
     virtual void set(size_t index, const core::Slice<uint8_t>& buffer) = 0;
 
     //! Repair source packet buffer.
     virtual core::Slice<uint8_t> repair(size_t index) = 0;
 
-    //! Reset current block.
-    virtual void reset() = 0;
+    //! Perform housekeeping for current block.
```

Comment:
I'd make these comments:

> Perform an initial setup for current block:

> Perform housekeeping for current block.

a bit more explicit:

```
Start block.

@remarks
 Performs an initial setup for a block. Should be called before <...>
```

```
Finish block.

@remarks
 Cleanups the resources allocated for the block. Should be called <...>
```

Reply:
Also note that this comment:

```
    //! Perform an initial setup for current block:
    //!  * Set number of source packets and repair packets.
```

does not fit well with the autobrief doxygen mode which we're using currently.

Reply:
I got it. Fixed